packages/core/src/StateNode.ts

```diff
@@ -1629,16 +1629,17 @@ class StateNode<
         (stateNode) => !(stateNode.type === 'history')
       );
     } else if (this.initial !== undefined) {
-      if (!this.states[this.initial]) {
+      if (!this.states[this.initial as string]) {
```

Comment:
TS became more conservative when it comes to `keyof T` in generic contexts. After all `keyof T`'s constraint is `PropertyKey` and that is equivalent to `string | number | symbol`. We cast this to `string` as that is what it's going to be in practice anyway.